test_agent/summarizer_agent/utils.py

```
@@ -0,0 +1,41 @@
+        genai.configure(api_key=api_key)
+        model = genai.GenerativeModel("gemini-2.5-flash-preview-05-20")
+        response = model.generate_content(prompt)
+        
+        # AI Agent Best Practice: Handle potentially empty or non-text responses from LLM
+        if response is None or not hasattr(response, 'text') or not response.text:
+            return "No meaningful text response was generated by the LLM."
+            
+        return response.text
+    except Exception as e:
+        # AI Agent Best Practice: Re-raise specific exceptions or provide context
+        raise Exception(f"Error calling Gemini API: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)